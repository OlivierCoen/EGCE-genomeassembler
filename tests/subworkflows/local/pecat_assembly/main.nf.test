nextflow_workflow {

    name "Test Workflow PECAT_ASSEMBLY"
    script "subworkflows/local/pecat_assembly/main.nf"
    workflow "PECAT_ASSEMBLY"
    tag "pecat_assembly"

    test("Small data") {

        when {

            params {
                pecat_config_file = "${projectDir}/tests/test_data/pecat/unzip/pecat_optional_config.txt"
            }

            workflow {
                """
                input[0] = [
                    [ id: "test", genome_size: 1500000 ],
                    file( '$projectDir/tests/test_data/pecat/unzip/reads.fasta.gz', checkIfExists: true)
                ]
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
