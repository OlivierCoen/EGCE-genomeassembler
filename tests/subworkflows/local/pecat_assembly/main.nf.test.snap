{
    "Small data": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "haplotype_1.fasta:md5,c122e83f6c3305ccbdeb2282c1b36925"
                    ]
                ],
                "1": [
                    
                ],
                "assembly_fasta": [
                    [
                        {
                            "id": "test",
                            "genome_size": 1500000
                        },
                        "haplotype_1.fasta:md5,c122e83f6c3305ccbdeb2282c1b36925"
                    ]
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-21T18:03:09.331832759"
    }
}